File: app/src/main/java/com/bignerdranch/android/mediafiles/discovery/worker/MediaChangeWorker.java
package com.bignerdranch.android.mediafiles.discovery.worker;

import android.content.Context;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.work.Worker;
import androidx.work.WorkerParameters;

import com.bignerdranch.android.mediafiles.MediaFilesApplication;

import javax.inject.Inject;

public class MediaChangeWorker extends Worker {
    @Inject protected ScanAddedTask scanAddedTask;
    @Inject protected ScanDeletedTask scanDeletedTask;
    @Inject protected WorkerSchedule workerSchedule;

    public MediaChangeWorker(@NonNull Context context, @NonNull WorkerParameters workerParams)
    {
        super(context, workerParams);
        MediaFilesApplication.getAppComponent().inject(this);
    }

    /**
     * 耗时的任务，在doWork()方法中执行
     * */
    @NonNull
    @Override
    public Result doWork()
    {
        Log.v("ShiyihuiHLNSKQ", "MediaChangeWorker doWork()");
        scanAddedTask.run();
        scanDeletedTask.run();

        workerSchedule.setupMediaStoreChangeWorker();

        return Result.success();
    }
}

File: app/src/main/java/com/bignerdranch/android/mediafiles/discovery/worker/MediaChangeWorker.kt
package com.bignerdranch.android.mediafiles.discovery.worker

import android.content.Context
import androidx.work.CoroutineWorker
import androidx.work.Worker
import androidx.work.WorkerParameters
import com.bignerdranch.android.mediafiles.DTAG
import com.bignerdranch.android.mediafiles.MediaFilesApplication
import com.bignerdranch.android.mediafiles.util.log.Logger
import javax.inject.Inject

class MediaChangeWorker(context: Context, workerParams: WorkerParameters) : CoroutineWorker(context, workerParams) {
    @Inject lateinit var scanAddedTask: ScanAddedTask
    @Inject lateinit var scanDeletedTask: ScanDeletedTask
    @Inject lateinit var workerSchedule: WorkerSchedule
    @Inject lateinit var logger: Logger

    /**
     * 耗时的任务，在doWork()方法中执行
     */
    override suspend fun doWork(): Result {
        logger.v(DTAG, "MediaChangeWorker doWork()")
        scanAddedTask.run()
        scanDeletedTask.run()
        workerSchedule.setupMediaStoreChangeWorker()
        return Result.success()
    }

    init {
        MediaFilesApplication.appComponent.inject(this)
    }
}
